### ISSUE TITLE
On the Measure of Intelligence

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#151]
## TL; DR

- Evaluation of specific skills does not lead our way to *general AI*.
- ***Intelligence*** is associated with the ability to learn new skills (with consideration to available prior and experience), and hence our goal for general AI should be specifically targeted at *the ability to learn* while being actionable (i.e., directly optimizable).

## Motivation

- Current research on ML and AI focus on mastering specific skills, sometimes beyond human levels.
- However, the pathway from specific to general is rather unclear compared to that from general to specific, hence direct analogy on assessing intelligence between human agents and artificial agents may potentially include fatal flaws.
- We need to thoroughly analyze human priors and evaluate *general AI agents* according to protocols that adequately reflect prior/experience/other conditions.

### Generally on Human Cognition

- Humans are neither **a collection of hard-coded priors from evolution** nor **blank slate that can learn anything with equal difficulty**. Rather we are in between the two extremes.
- Evolution has equipped us with a set of **natural priors** that facilitates our learning on relevant tasks/skills.
- Evolution has mostly focused on *adaptability* and the ability to *pick up new and relevant skills*.
- Humans are relatively general reasoning agents, but our generality has clear limits. (4+dimension, environments other than Earth)

### Human Priors

- Low-level *sensorimotor* priors (senses, body movement, reflexes)
- Meta-learning Priors (causality, spatio-temporal continuity, familiarity to modular-hierarchical structure)
- High-level *knowledge* priors (objectness, agentness, natural numbers, geometry)

### Human Knowledge Priors

- Objectness: Environments should be parsed into *objects* characterized by principles of cohesion, persistence and contact
- Agentness: Animate objects have intentions of their own and pursue goals of some sort
- Natural Number: *entities observed through any sensory modality*
- Geometry: distance and orientation

### Overall

- Intelligence lies in broad or general-purpose abilities (flexibility + adaptability / skill acquisition and generalization)
- The history of AI research has been ***a slow climb*** along the spectrum of generalization. (no generality - local - broad - full)
- Measure of intelligence should control for **experience** and **prior** and should seek to quantify generalization power.
- General AI should be benchmarked against human intelligence and should be founded on a similar set of knowledge priors.

### ARC dataset

- Stay close to psychometric intelligence tests formats
- Focus on developer-aware generalization (system developer should also be ignorant about specific tasks to be measured upon)
- Measure broad form of generalization (across tasks sharing a broad action category)
- Quantitatively control experience (training samples / time)
- Explicitly describe the complete set of priors (control priors needed for the test)

## Comments and thoughts

- First of all, a very well written paper. Big congrats to the author. Great pleasure reading it.

### One-way path from General to Specific

- The idea that building an intelligent artificial agent that beats humans on specific but *complex enough* task such as chess or GO really taught us ***nothing*** about general intelligence is quite shocking. (AlphaGO has literally no other application than playing GO)
- Humans regard someone who really excels at GO or chess *intelligent* for two reasons: (1) We share almost identical natural priors (and we are not artificial agents specifically designed to play chess or GO) (2) We assume that we've had similar experience (it's no shame losing to someone who's devoted his whole life in chess as a normal person)
- However, these fundamental assumptions fail for artificial agents, hence the fact that they are better at some complex task than humans confirms next to nothing in terms of general intelligence.
- To extend the discussion, setting benchmarks on specific (but complex) tasks may not be helpful for building agents with broad intelligence.

### Psychometrics

- I've never heard of this subfield of psychology up until now. Glad that I've come across now.
- A professor in statistics has taught me that factor analysis originated from analysis on human intelligence (such as IQ). This preliminary helped. 
- Hence, psychometrics has its root on statistics, such as eigenvalue decomposition and factor analysis (analysis on covariance)
- To measure intelligence (a latent factor high up in the pyramid), psychometrists collect test results on diverse tasks (skills) and measure the latent factor via statistical modeling (such as factor analysis). This really makes sense, and makes me wonder why such practice has not been explored much in the field of AI.

### Human Evolutionary Priors

- 4 central knowledge priors: objectness, agentness, natural number and geometry. These all makes sense, and wonder who has studied such challenging field of research.
- The idea that human intelligence is *generally general* but has limitations (in that it only helps for environments similar to Earth and so on) is really persuasive, and I side with the author on his claim that our general AI should aim for benchmarking human intelligence (since it's the only form of intelligence we have adequate access for)
- General intelligence is a *spectrum*.

### ARC benchmark

- Clever conception and reasonable implementation.
